qfieldsync/core/cloud_converter.py

```diff
@@ -97,7 +97,25 @@ def convert(self) -> None:  # noqa: C901
                     if layer_source.is_localized_path:
                         continue
 
-                if layer.type() == QgsMapLayer.VectorLayer:
+                if layer.dataProvider().name() == "virtual":
```

Comment:
```suggestion
            if layer.type() == QgsMapLayer.VectorLayer:
                if layer.dataProvider().name() == "virtual":
                	# ...
                else:
                	# ...
```
Since all the virtual layers are also vector layers, can we have it within that `if`. I know it adds a bit of indentation, but it skips a "mental check" why it is outside.

Also note you have some unfortunate cases when you don't have data provider, not sure how do you handle such situations.